src/libcprover-rust/src/lib.rs

```diff
@@ -87,8 +99,17 @@ mod tests {
         // Validate integrity of passed goto-model.
         client.validate_goto_model();
 
+        // ATTENTION: The following `.clone()` is unneeded - I keep it here in order to
+        // make potential printing of the message buffer easier (because of borrowing).
+        // This is also why a print instruction is commented out (as a guide for someone
+        // else in case they want to inspect the output).
+        let validation_msg = "Validating consistency of goto-model supplied to API session";
         let msgs = ffi::get_messages();
-        print_response(msgs);
+        let msgs_assert = translate_response_buffer(msgs).clone();
+
+        assert!(msgs_assert.contains(&String::from(validation_msg)));
```

Comment:
Yeah, this is very hard to do right now - the C++ API returns `void`, and that's also because the function it's delegating to, `validate_goto_model` in `src/goto-programs/validate_goto_model.cpp:128` is also `void`.

This will require some deep refactoring for these to behave this way.

The assertions I have added are along the lines of "Test for behaviour - not interface", which are just making sure that a side-effect of the action has been performed accurately.

Not terribly robust tests, but better than nothing - until at least a better interface presents itself.

Reply:
Understood that this is test code and proper result structures are future work.